src/main/java/botmanager/Parser.java

```diff
@@ -0,0 +1,122 @@
+package botmanager;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import action.Action;
+import action.AddTaskAction;
+import action.DeleteTaskAction;
+import action.FindTasksAction;
+import action.HelpAction;
+import action.ListTasksAction;
+import action.MarkTaskAction;
+import action.UnmarkTaskAction;
+import enums.Command;
+import exception.BotException;
+import exception.InvalidCommandException;
+import exception.InvalidCommandFormatException;
+import exception.InvalidDateFormatException;
+import exception.InvalidTaskIndexException;
+import task.Deadline;
+import task.Event;
+import task.Todo;
+
+/**
+ * Class to handle parsing of user input.
+ *
+ * @author dwsc37
+ */
+public class Parser {
+    /**
+     * Parses a user input string and returns an Action to be executed.
+     *
+     * @param input User input string.
+     * @return Action to be executed by BotManager.
+     * @throws BotException If the command is not recognised or formatted wrongly.
+     */
+    public Action parseInput(String input) throws BotException {
+        Command command = Command.getCommandFromInput(input.split(" ")[0]);
+        return switch (command) {
+        case LIST -> new ListTasksAction();
```

Comment:
Great job on this! Many users make the mistake of indenting the case statements.